### PR TITLE
Fitting improvements

### PR DESCRIPTION
**Pull request by mdecleir**
I updated the boundaries of the fitting parameters, and added weights to the fitting.

### CODE REVIEW

**Comment by mdecleir** [comment]
I updated and re-arranged the power law fitting and plotting functions. Fitting can now be done both with SpeX spectra and band data, and with data in E(lambda-V) or A(lambda)/A(V) units.

**Comment by mdecleir** [comment]
@karllark The python test for code style fails because fo the following reason:

measure_extinction/extdata.py:1094:13: E731 do not assign a lambda expression, use a def
measure_extinction/extdata.py:1096:13: E731 do not assign a lambda expression, use a def

What is your opinion on using lambda functions?

**Comment by mdecleir** [comment]
@karllark , I am also a little annoyed by the very long array in test_extdata.py to test the results of the fitting. Any suggestion on how to avoid this?

**Comment by karllark** [comment]
> What is your opinion on using lambda functions?

I have been moving away from them.  It is usually just as simple to make to define a function at the same point in the code as you would do a lambda function.  It takes two lines instead of one only.

File: measure_extinction/extdata.py [hunk 1]
```diff
@@ -1056,38 +1062,105 @@ def plot(
...
+        """
+        # retrieve the band data to be fitted
+        ftype = "BAND"
+        gbool = np.all(
+            [
+                (self.npts[ftype] > 0),
+                (self.waves[ftype] > 1.0 * u.micron),
+                (self.waves[ftype] < 40.0 * u.micron),
+            ],
+            axis=0,
```
Comment (karllark) [inline measure_extinction/extdata.py:1087]: This is very specific to SpeX.  *After* you have your talk done, I would suggest that this code is a good candidate for moving to a separate repository specifically for the SpeX project.

Comment (karllark) [reply measure_extinction/extdata.py:1087]: Really this whole function given it also has a model that is specific to the NIR fitting.

File: measure_extinction/tests/test_extdata.py [hunk 2]
```diff
@@ -96,3 +96,3115 @@ def test_get_column_val():
...
+
+    # test the fitting results
+    waves = np.sort(extdata.get_fitdata(["SpeX_SXD", "SpeX_LXD"])[0].value)
+    np.testing.assert_almost_equal(extdata.model["waves"], waves)
+    exts = [
+        -1.21495627,
+        -1.21631954,
+        -1.21768176,
+        -1.21904232,
+        -1.22040163,
```
Comment (karllark) [inline measure_extinction/tests/test_extdata.py:163]: For this long of an array of test data, I would put it in an ascii file in a data subdir and then just read it in.  It is small enough that that it is fine to have as a part of the respository.

**Comment by karllark** [comment]
I am find merging this PR.  We can discuss this later this week before I do so.  Otherwise, tell me and I'll merge it now.

**Comment by mdecleir** [comment]
> I have been moving away from them. It is usually just as simple to make to define a function at the same point in the code as you would do a lambda function. It takes two lines instead of one only.

Ok, I replaced the lambda with a def function at the same point in the code, but this means that there are function definitions inside a function. Is that ok?

File: measure_extinction/extdata.py (hunk 1 above)
Comment (mdecleir) [reply measure_extinction/extdata.py:1087]: Ok. I opened an issue for this: #68

File: measure_extinction/tests/test_extdata.py (hunk 2 above)
Comment (mdecleir) [reply measure_extinction/tests/test_extdata.py]: Done!

**Comment by mdecleir** [comment]
All checks have passed now, so I think this PR can be merged.

**Comment by karllark** [comment]
> > I have been moving away from them. It is usually just as simple to make to define a function at the same point in the code as you would do a lambda function. It takes two lines instead of one only.
> 
> Ok, I replaced the lambda with a def function at the same point in the code, but this means that there are function definitions inside a function. Is that ok?

Yep.

**Review by karllark — APPROVED** [review]
Looks good.

**Pull request merged**